pkg/kubelet/images/image_manager_test.go

```diff
@@ -153,20 +155,32 @@ func pullerTestCases() []pullerTestCase {
 			qps:        2000.0,
 			burst:      0,
 			expected: []pullerExpects{
-				{[]string{"GetImageRef"}, ErrImagePull},
-				{[]string{"GetImageRef"}, ErrImagePull},
-				{[]string{"GetImageRef"}, ErrImagePullBackOff},
+				{[]string{"GetImageRef"}, ErrImagePull, true, false},
+				{[]string{"GetImageRef"}, ErrImagePull, true, false},
+				{[]string{"GetImageRef"}, ErrImagePullBackOff, false, false},
 			}},
 	}
 }
 
-type mockPodPullingTimeRecorder struct{}
+type mockPodPullingTimeRecorder struct {
+	startedPullingRecorded  bool
+	finishedPullingRecorded bool
+}
 
-func (m *mockPodPullingTimeRecorder) RecordImageStartedPulling(podUID types.UID) {}
+func (m *mockPodPullingTimeRecorder) RecordImageStartedPulling(podUID types.UID) {
+	m.startedPullingRecorded = true
```

Comment:
would it make sense to guard the fields with locks, to avoid race condition, like fake_runtime? Ref: https://github.com/kubernetes/kubernetes/blob/10639a7963b31f43765d2abf53aa946fd1123bcc/pkg/kubelet/container/testing/fake_runtime.go#L115-L116